tests/kani/Intrinsics/FastMath/add.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Check that `fadd_fast` overflow checks pass with suitable assumptions
+
+#![feature(core_intrinsics)]
+
+fn main() {
+    let x: f64 = kani::any();
+    let y: f64 = kani::any();
+
+    kani::assume(x.is_finite());
+    kani::assume(y.is_finite());
+    match (x.is_sign_positive(), y.is_sign_positive()) {
+        (true, true) => kani::assume(x < f64::MAX - y),
+        (false, false) => kani::assume(x > f64::MIN - y),
+        _ => (),
+    }
+    let _z = unsafe { std::intrinsics::fadd_fast(x, y) };
```

Comment:
Can you please assert that the result is equal to regular add? The same for the other tests.

Reply:
Changed it so we assert it in the case of addition and substraction, as other operations show serious performance issues. Created #809 to revisit this.